### ISSUE TITLE
Issue with Reconstructing Real Scene from Vehicle Camera Images

### ISSUE THREAD

**Issue by msanchezvicom** [nerfstudio-project/nerfstudio#2151]
Hello,

I am currently facing an issue while attempting to reconstruct a real scene captured by a camera attached to a vehicle. To accomplish this, I am running the ns-process command with the following parameters:
`ns-process-data images --camera-type fisheye --data {INPUT_PATH} --output-dir {OUTPUT_PATH}`

The camera poses are well approximated:
![image](https://github.com/nerfstudio-project/nerfstudio/assets/112627154/837ae023-8d25-4da6-9f8c-e2fa7097df27)

Additionally, since I am working with fish-eye images, I have set up a mask and specified it in the transforms.json file:

![image](https://github.com/nerfstudio-project/nerfstudio/assets/112627154/fb78e13c-c45f-4258-b728-3e530f49fde4)

Next, I proceed to use nerfacto for training with the following command:
`ns-train nerfacto --data {OUTPUT_PATH} --pipeline.datamanager.camera-optimizer.mode off`

The training phase starts successfully, as shown in the following image:

![image](https://github.com/nerfstudio-project/nerfstudio/assets/112627154/9334dbe3-7f72-4d60-9c45-3afc34b85279)

However, after a few iterations (approximately 6-10%), the reconstructed scene disappears from the render and does not reappear again:

![image](https://github.com/nerfstudio-project/nerfstudio/assets/112627154/f7c0585d-b9bd-410b-902d-b29537f021c0)

I am unsure of what I might be doing wrong in this process. I would greatly appreciate any recommendations or insights that could help me resolve this issue.

Thank you very much for your assistance!

**Comment by tancik** [comment]
This can happen when the training diverges and produces a NAN breaking everything. It is hard to say exactly why that is happening in this case (sometimes it is pretty random). It does look like colmap did fail though given that one of the images is not along the path, there is a slight chance that this messes everything up.

**Comment by msanchezvicom** [comment]
Hi @tancik ,

Thanks for your reply. I managed to remove the problematic image from the transforms.json file, but I'm still facing issues with the reconstructed scene:

![Screenshot from 2023-07-03 09-08-49](https://github.com/nerfstudio-project/nerfstudio/assets/112627154/bdb66202-8219-4d0d-8b63-e5038a75ac50)

I'd appreciate it if you could share any additional insights or suggestions to help me solve this problem.

Thank you very much

**Comment by msanchezvicom** [comment]
Hi @tancik ,

I have calculated the poses of my camera by considering both the odometry of my vehicle and the camera's pose to ensure they are accurately set and that the images are in the correct order. Additionally, I have mapped these poses to OpenStreetMaps to verify that the vehicle followed the intended path:
![path_car_1](https://github.com/nerfstudio-project/nerfstudio/assets/112627154/40a611fe-cedb-416e-9da9-f622d91541b9)

However, despite training the model without encountering any error messages, I am still experiencing the same issue as mentioned in my previous messages. The scene is not being reconstructed as expected:

![image](https://github.com/nerfstudio-project/nerfstudio/assets/112627154/23f8d6d7-a935-4976-9691-3102aaf0f920)


I would greatly appreciate your assistance in resolving this issue.

Thank you very much!

**Comment by msanchezvicom** [comment]
Hi @tancik ,

I have some good news to share! I decided to train **instant-ngp** instead of **nerfacto**, and I'm happy to report that the scene is now being reconstructed correctly. I'm not entirely sure why this change made a difference, but it seems to be working well.


https://github.com/nerfstudio-project/nerfstudio/assets/112627154/ba0dff1d-1bfa-4b70-8a11-b0f31d9ac4c4



Next, I plan to properly hypertune the model to achieve even better results, as there are still some artifacts present. Nevertheless, I wanted to express my gratitude for your work and support, both on this specific issue and on other issues I have posted. Thank you!
